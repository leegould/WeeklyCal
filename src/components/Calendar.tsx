import React, { PureComponent } from 'react';
import { View } from 'react-native';
import { Calendar } from 'react-native-calendars';
import moment from 'moment';
import { NavigationActions } from 'react-navigation';
import { WeekState } from '../types';

type Props = {
    navigation: {
        dispatch: Function,
    },
    data: WeekState,
    onChangeDate: Function,
};

export default class CalendarModal extends PureComponent<Props> {
    static navigationOptions = {
        headerTitle: 'Week Start',
    }

    render() {
        return (
            <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                <Calendar
                    current={this.props.data.week.days[0].date.format('YYYY-MM-DD')}
                    onDayPress={(day: any) => {
                        // console.log('Calendar.onDayPress', day)
                        this.props.onChangeDate(moment(day.dateString));
                        const backAction = NavigationActions.back();
                        this.props.navigation.dispatch(backAction);
                    }}
                />
            </View>
        );
    }
}
