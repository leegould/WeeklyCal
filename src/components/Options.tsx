import React from 'react';
import { View, Switch, StyleSheet, Animated } from 'react-native';
import { ListItem } from 'react-native-elements';
import moment from 'moment';
import { CalendarsState, Calendar } from '../types';

type Props = {
    navigation: {
        navigate: Function,
    }
    data: CalendarsState,
    onFetchCalendars: Function,
    onChangeDate: Function,
    onToggleShowAll: Function,
    onToggleCalendar: Function,
};

type State = {
    anim: Animated.Value,
};

export default class Options extends React.PureComponent<Props, State> {
    static navigationOptions = {
        title: 'Options',
    };

    constructor(props: Props) {
        super(props);

        this.state = {
            anim: new Animated.Value(this.props.data.showAll ? 0 : 1),
        };

        try {
            props.onFetchCalendars();
        }
        catch (err) {
            console.log('Options.componentDidMount.error', err);
        }
    }

    toggleAllDay() {
        if (this.props && this.props.data.showAll) {
            this.show();
        } else {
            this.hide();
        }
        this.props.onToggleShowAll();
        this.props.onChangeDate(moment(), this.props.data.showAll, this.props.data.selectedCalendars);
    }

    toggleCalendar(item: Calendar) {
        this.props.onToggleCalendar(item);
        // TODO : how to get selected date? pass in?
        this.props.onChangeDate(moment(), this.props.data.showAll, this.props.data.selectedCalendars);
    }

    show() {
        Animated.timing(this.state.anim, {
            toValue: 1,
            duration: 250,
        }).start()
    }

    hide() {
        Animated.timing(this.state.anim, {
            toValue: 0,
            duration: 250,
        }).start()
    }


    render() {
        // console.log('Options.render.data', this.props.data);

        return (
            <View style={{ flex: 1,  backgroundColor: 'gray' }}>
                {!this.props.data.isFetching &&
                <View style={{ flex: 1, marginVertical: 5 }}>
                    <ListItem
                        key={'showAll'}
                        title={'Show All Calendars'}
                        rightAvatar={
                            <Switch
                                onValueChange={() => this.toggleAllDay()}
                                value={this.props.data.showAll}
                                trackColor={{true: '#C2272D', false: ''}}
                                style={styles.switchInput}
                            />
                        }
                        containerStyle={{ backgroundColor: 'lightgray', borderBottomWidth: 1, borderBottomColor: 'gray' }}
                        titleStyle={{ color: 'white' }}
                        subtitleStyle={{ color: 'white' }}
                    />
                    {!this.props.data.showAll &&
                    <Animated.FlatList
                        style={{ opacity: this.state.anim }}
                        data={this.props.data.allCalendars}
                        scrollEnabled={false}
                        renderItem={({item}: {item: Calendar}) => {
                                return(
                                    <ListItem
                                        key={item.id}
                                        title={item.title}
                                        subtitle={item.source}
                                        rightAvatar={
                                            <Switch
                                                onValueChange={() => this.toggleCalendar(item)}
                                                // this.props.data.selectedCalendars.findIndex(x => x.id === item.id) > -1
                                                value={this.props.data.selectedCalendars.indexOf(item.id) > -1}
                                                // trackColor={{true: '#C2272D', false: ''}}
                                                trackColor={{true: item.color, false: item.color}}
                                                style={styles.switchInput}
                                            />
                                        }
                                        containerStyle={{ backgroundColor: 'lightgray', borderBottomWidth: 0.5, borderBottomColor: 'gray' }}
                                        titleStyle={{ color: 'white' }}
                                        subtitleStyle={{ color: 'white' }}
                                    />
                                );
                            }
                        }
                        keyExtractor={(item: Calendar, index: number) => item.id}
                    />
                    }
                </View>
                }
            </View>
        );
    }
}

const styles = StyleSheet.create({
    switchInput: {
        transform: [{ scaleX: .8 }, { scaleY: .8 }]
    },
});
