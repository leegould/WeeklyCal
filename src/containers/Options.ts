import { connect } from 'react-redux';
import { Moment } from 'moment';
import Options from '../components/Options';
import {
    fetchCalendars,
    changeWeekDate,
    toggleShowAllAndUpdateWeek,
    toggleCalendarAndUpdateWeek,
    toggleResetDateOption,
    toggleEventColorOptionAndUpdateWeek,
    toggleDayAddLinkAndUpdateWeek,
    toggleRollingWeekOption,
    toggleInlineAddOption,
} from '../actions';
import { Calendar } from '../types';

const mapStateToProps = (state: any) => {
    return { 
        data: state.calendars,
        options: state.options,
    };
};

const mapDispatchToProps = (dispatch: any) => ({
    onFetchCalendars: () => {
        dispatch(fetchCalendars());
    },
    onChangeDate: (date: Moment, showAll: boolean, selectedCalendars: string[], rollingWeek: boolean) => {
        dispatch(changeWeekDate(date, showAll, selectedCalendars, rollingWeek));
    },
    onToggleShowAllAndUpdateWeek: () => {
        dispatch(toggleShowAllAndUpdateWeek());
    },
    onToggleCalendarAndUpdateWeek: (calendar: Calendar) => {
        dispatch(toggleCalendarAndUpdateWeek(calendar));
    },
    onToggleResetDateOption: () => {
        dispatch(toggleResetDateOption());
    },
    onToggleEventColorOptionAndUpdateWeek: () => {
        dispatch(toggleEventColorOptionAndUpdateWeek());
    },
    onToggleDayAddLinkAndUpdateWeek: () => {
        dispatch(toggleDayAddLinkAndUpdateWeek());
    },
    onToggleRollingWeek: () => {
        dispatch(toggleRollingWeekOption());
    },
    onToggleInlineAdd: () => {
        dispatch(toggleInlineAddOption());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Options);