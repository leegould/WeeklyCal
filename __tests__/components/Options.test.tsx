import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import { Options } from '../../src/components/Options';

jest.useFakeTimers();

test('renders correctly with expected args', () => {
    const props = {
        navigation: {
            getParam: () => {},
            navigate: () => {},
        },
        date: moment('25/12/2005', 'DD/MM/YYYY'),
        options: {
            resetDate: false,
            eventColor: false,
            dayAddLink: false,
            rollingWeek: false,
            inlineAdd: false,
            eventRowBorder: false,
        },
        data: {
            isFetching: false,
            showAll: true,
            allCalendars: [],
            selectedCalendars: [],
        },
        onFetchCalendars: () => {},
        onChangeDate: () => {},
        onToggleShowAllAndUpdateWeek: () => {},
        onToggleCalendarAndUpdateWeek: () => {},
        onToggleResetDateOption: () => {},
        onToggleEventColorOptionAndUpdateWeek: () => {},
        onToggleDayAddLinkAndUpdateWeek: () => {},
        onToggleRollingWeekAndUpdateWeek: () => {},
        onToggleInlineAddAndUpdateWeek: () => {},
        onToggleEventRowBorderAndUpdateWeek: () => {},
    }

    const tree = renderer.create(
        <Options {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
