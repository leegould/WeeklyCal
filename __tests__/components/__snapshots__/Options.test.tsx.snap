// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with expected args 1`] = `
<View
  style={
    Object {
      "backgroundColor": "gray",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 5,
      }
    }
  >
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Allow Date Reset
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Hold down the calendar picker in the header to reset to todays date
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Event Colours
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Display the calendar colour next to the event
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Add Link Per Day
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Choose if each day should display an add event shortcut
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Rolling Week
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              The first day shown of the week should be the selected day
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Inline Add
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Allow quick adding of an event for each day
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Event Line Separators
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Divide events in each day with a line
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={false}
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightgray",
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 17,
                }
              }
              testID="listItemTitle"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Show All Calendars
            </Text>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginVertical": 5,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Display events from all calendars, or choose specific ones
            </Text>
          </View>
          <View
            width={16}
          />
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#C2272D"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 0.8,
                    },
                    Object {
                      "scaleY": 0.8,
                    },
                  ],
                },
              ]
            }
            tintColor="#C2272D"
            value={true}
          />
        </View>
      </View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "opacity": 0,
          }
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
